File: v131016/TeeNew/Tool_AxisScroll.pas
unit Tool_AxisScroll;
{$I TeeDefs.inc}

interface

uses
  Windows, Messages,
  SysUtils, Classes,
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls,
  Base, TeEngine, TeeTools, Series, TeeProcs, Chart;

type
  TAxisScrollToolDemo = class(TBaseForm)
    Series1: TFastLineSeries;
    ChartTool1: TAxisScrollTool;
    CheckBox1: TCheckBox;
    CheckBox2: TCheckBox;
    procedure CheckBox1Click(Sender: TObject);
    procedure CheckBox2Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$R *.dfm}

procedure TAxisScrollToolDemo.CheckBox1Click(Sender: TObject);
begin
  ChartTool1.Active:=CheckBox1.Checked;
end;

procedure TAxisScrollToolDemo.CheckBox2Click(Sender: TObject);
begin
  ChartTool1.ScrollInverted:=CheckBox2.Checked;
end;

initialization
  RegisterClass(TAxisScrollToolDemo);
end.

File: TeeNew/Builder2009/Tool_AxisScroll.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Tool_AxisScroll.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeeTools"
#pragma resource "*.dfm"
TAxisScrollToolDemo *AxisScrollToolDemo;
//---------------------------------------------------------------------------
__fastcall TAxisScrollToolDemo::TAxisScrollToolDemo(TComponent* Owner)
  : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TAxisScrollToolDemo::CheckBox1Click(TObject *Sender)
{
  ChartTool1->Active = CheckBox1->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TAxisScrollToolDemo::CheckBox2Click(TObject *Sender)
{
  ChartTool1->ScrollInverted = CheckBox2->Checked;
}
//---------------------------------------------------------------------------

